src/libraries/Microsoft.Extensions.FileProviders.Physical/tests/PhysicalFileProviderTests.netcoreapp.cs

```diff
@@ -0,0 +1,146 @@
+        }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public void UsePollingFileWatcher_UseActivePolling_HasChanged_SymbolicLink_TargetNotExists(bool useWildcard)
+        {
+            // Arrange
+            using var rootOfLink = new DisposableFileSystem();
+            string linkName = Path.GetRandomFileName();
+            string linkPath = Path.Combine(rootOfLink.RootPath, linkName);
+            File.CreateSymbolicLink(linkPath, "not-existent-file");
+
+            // Act
+            using var provider = new PhysicalFileProvider(rootOfLink.RootPath) { UsePollingFileWatcher = true, UseActivePolling = true };
+            IChangeToken token = provider.Watch(useWildcard ? "*" : linkName);
+
+            // Assert
+            Assert.False(token.HasChanged);
+        }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public void UsePollingFileWatcher_UseActivePolling_HasChanged_SymbolicLink_TargetChanged(bool useWildcard)
+        {
+            // Arrange
+            using var rootOfFile = new DisposableFileSystem();
+            string file1Path = Path.Combine(rootOfFile.RootPath, Path.GetRandomFileName());
+            File.WriteAllText(file1Path, "v1.1");
+
+            string file2Path = Path.Combine(rootOfFile.RootPath, Path.GetRandomFileName());
+            File.WriteAllText(file2Path, "v2.1");
+
+            using var rootOfLink = new DisposableFileSystem();
+            string linkName = Path.GetRandomFileName();
+            string linkPath = Path.Combine(rootOfLink.RootPath, linkName);
+            File.CreateSymbolicLink(linkPath, file1Path);
+
+            string filter = useWildcard ? "*" : linkName;
+            using var provider = new PhysicalFileProvider(rootOfLink.RootPath) { UsePollingFileWatcher = true, UseActivePolling = true };
+            IChangeToken token = provider.Watch(filter);
+            Assert.False(token.HasChanged);
+
+            // Act 1 - Change file 1's content.
+            File.WriteAllText(file1Path, "v1.2");
+            Thread.Sleep(GetTokenPollingInterval(token));
+
+            // Assert 1
+            Assert.True(token.HasChanged);
+
+            // Act 2 - Change link target to file 2.
+            token = provider.Watch(filter); // Once HasChanged is true, the value will always be true. Get a new change token.
+            Assert.False(token.HasChanged);
+            File.Delete(linkPath);
+            File.CreateSymbolicLink(linkPath, file2Path);
+            Thread.Sleep(GetTokenPollingInterval(token));
+
+            // Assert 2
+            Assert.True(token.HasChanged); // It should report the change regardless of the timestamp being older.
+
+            // Act 3 - Change file 2's content.
+            token = provider.Watch(filter);
+            Assert.False(token.HasChanged);
+            File.WriteAllText(file2Path, "v2.2");
+            Thread.Sleep(GetTokenPollingInterval(token));
```

Comment:
Again I think this test isn't necessary as it overlaps other tests you already do.